sopel/tools/memories.py

```diff
@@ -166,3 +205,46 @@ def __contains__(self, key):
 
     def __setitem__(self, key: Optional[str], value):
         super().__setitem__(self._make_key(key), value)
+
+    def setdefault(self, key: str, default=None):
+        return super().setdefault(self._make_key(key), default)
+
+    def __delitem__(self, key: str):
+        super().__delitem__(self._make_key(key))
+
+    def copy(self):
+        return self.__class__(
+            self,
+            identifier_factory=self.make_identifier,
+        )
+
+    def get(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().get(self._make_key(key))
+        return super().get(self._make_key(key), default)
+
+    def pop(self, key: str, default=_NO_DEFAULT):
+        if default is _NO_DEFAULT:
+            return super().pop(self._make_key(key))
+        return super().pop(self._make_key(key), default)
+
+    def update(self, maybe_mapping=tuple()):
+        super().update(self._process_arg(maybe_mapping))
+
+    def __or__(self, other):
+        if not isinstance(other, dict):
+            return NotImplemented
+        new = type(self)(self, identifier_factory=self.make_identifier)
```

Comment:
Is it possible to use `copy()` here? If not, we should probably use the same spelling of referencing the type (i.e. use `type(self)` in both places or `self.__class__` in both places, but not mix the two)

```suggestion
        new = self.copy()
```

Reply:
> Do you place higher value on using `self.copy()` here because it's shorter, or on using `type(self)()` for parity with the sister method?

I would prefer `self.copy()` because it's more clearly expressing what's going on _and_ avoiding repetition.